scripts/scil_project_streamlines_to_map.py

```diff
@@ -0,0 +1,232 @@
+
+    p.add_argument('in_bundle',
+                   help='Fiber bundle file.')
+    p.add_argument('out_folder',
+                   help='Folder where to save endpoints metric.')
+
+    p1 = p.add_mutually_exclusive_group(required=True)
+    p1.add_argument('--in_metrics', nargs='+', default=[],
+                    help='Nifti metric(s) to compute statistics on.')
+    p1.add_argument('--use_dps', metavar='DPS_KEY', nargs='+',
+                    help='Use the data_per_streamline (scalar) from file, '
+                         'e.g. commit_weights.')
+    p1.add_argument('--use_dpp', metavar='DPP_KEY', nargs='+', default=[],
+                    help='Use the data_per_point (scalar) from file.')
+    p1.add_argument('--load_dps', metavar='DPS_KEY', nargs='+', default=[],
+                    help='Load data per streamline (scalar) .txt or .npy.')
+    p1.add_argument('--load_dpp', metavar='DPP_KEY', nargs='+', default=[],
+                    help='Load data per point (scalar) from .txt or .npy.')
+
+    p.add_argument('--from_wm', action='store_true',
+                   help='Project metrics from whole streamlines coverage.')
+    p.add_argument('--to_wm', action='store_true',
+                   help='Project metrics into streamlines coverage.')
+    add_reference_arg(p)
+    add_overwrite_arg(p)
+    return p
+
+
+def _compute_streamline_mean(cur_ind, cur_min, cur_max, data):
+    # From the precomputed indices, compute the binary map
+    # and use it to weight the metric data for this specific streamline.
+    cur_range = tuple(cur_max - cur_min)
+    streamline_density = compute_tract_counts_map(ArraySequence([cur_ind]),
+                                                  cur_range)
+    streamline_data = data[cur_min[0]:cur_max[0],
+                           cur_min[1]:cur_max[1],
+                           cur_min[2]:cur_max[2]]
+    streamline_average = np.average(streamline_data,
+                                    weights=streamline_density)
+    return streamline_average
+
+
+def _process_streamlines(streamlines, just_endpoints):
+    # Compute the bounding boxes and indices for all streamlines.
+    # just_endpoints will get the indices of the endpoints only for the
+    # usecase of projecting GM metrics into the WM.
+    mins = []
+    maxs = []
+    offset_streamlines = []
+
+    # Offset the streamlines to compute the indices only in the bounding box.
+    # Reduces memory use later on.
+    for idx, s in enumerate(streamlines):
+        mins.append(np.min(s.astype(int), 0))
+        maxs.append(np.max(s.astype(int), 0) + 1)
+        if just_endpoints:
+            s = np.stack((s[0, :], s[-1, :]), axis=0)
+        offset_streamlines.append((s - mins[-1]).astype(np.float32))
+
+    offset_streamlines = ArraySequence(offset_streamlines)
+    indices = uncompress(offset_streamlines)
+
+    return mins, maxs, indices
+
+
+def _project_metrics(curr_metric_map, count, orig_s, streamline_mean,
```

Comment:
This could also go into the library. I would also add comments to tell the user that this function has side-effects instead of returns.